### ISSUE TITLE
OAuth2 Invalid Credentials Error

### ISSUE THREAD

**Issue by abertay-university** [directus/directus#3831]
1) The issue and what you expected to happen
    OAuth authentication gets an INVALID_CREDENTIALS response after authenticating with the 3rd party service. The expected behavior is to match on the user email address, bypass password checks (because there is none) and create a sign in token.

2) Exact steps to reproduce this issue
    Set up OAuth2 provider in the environment variables. Create user with email address that matches the 3rd party service, (preferably without a password to prevent normal login). Go to login screen and click "Log in with <Provider>". Authenticate successfully with the 3rd party service. The redirect to the callback URI gets the INVALID_CREDENTIALS exception.

3) Your environment:
    MySQL 5.7
    npx and nginx
    Chrome 87 and Firefox 84.

4) Any other relevant information we might need to reproduce this issue
    With some manual debugging, the issue appeared after a change to the authentication.ts script.
    **Commit ID: 99052ad Treat no password as invalid**
```
		if (!password || !user.password) {
			throw new InvalidCredentialsException();
		}
```
If I comment out the code above, Oauth2 works again. There needs a bypass if the authentication attempt is from an oauth provider.

**Comment by megamorf** [comment]
+1 For this feature. As it is now, new users that signed in via OIDC will get an error message because their user account in Directus is not activated:

```json
{"errors":[{"message":"Invalid user credentials.","extensions":{"code":"INVALID_CREDENTIALS"}}]}
```

**Comment by zhibirc** [comment]
It's strange regarding the discussion above (seems, it had already fixed), but I still getting this error (`INVALID_CREDENTIALS`), despite:

1. OIDC in Okta is properly setup and it works for other application with identical credentials (both on localhost behind the Nginx reverse proxy).
2. Email address of that user was added to Directus.
3. User is active.

Could you please provide me with advice what are the possible reasons of such behaviour?

Also such message is appeared after unsuccessful logging:

<img width="364" alt="Screenshot 2023-01-18 at 12 41 10" src="https://user-images.githubusercontent.com/8087652/213151021-8be9269f-61d7-4bd1-a510-c88e9304ffad.png">

**Comment by abertay-university** [comment]
Hi @zhibirc,

There is a unique identifier that is required on the account in Directus that links it to the 3rd party authentication. I suspect the account in question doesn't have that. I find I need to delete the user in Directus and let the Okta create the user when they log in for the first time. You can configure how accounts are created by default in the .env file.

Otherwise you can manually add the identifier to the user account in Directus but that's not ideal
![image](https://user-images.githubusercontent.com/31690037/213153186-9a21c1f9-7220-4bc1-94e6-ca4305e94b03.png)

**Comment by zhibirc** [comment]
@abertay-university thank you very much!

I already tried to use the external identifier which I guess should be ClientID, but result was the same, unfortunately. But you helped to focus on this aspect a bit more. :)

When I more attentively look on the configuration I realized what I did incorrectly. There is an option in standard setup -- `AUTH_OKTA_IDENTIFIER_KEY="email"`. This option, apparently, is the one which used for 
determining what kind of value should be in "External Identifier" field. Because I have an "email" value for this option, I used user's email also as a value for this field, and successfully logged in without any errors.

Thanks again!
